libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProvider.kt

```diff
@@ -0,0 +1,35 @@
+package com.mapbox.navigation.core.routealternatives
+
+import com.mapbox.navigation.core.RouteProgressData
+
+internal object AlternativeRouteProgressDataProvider {
+
+    fun getRouteProgressData(
+        primaryRouteProgressData: RouteProgressData,
+        alternativeMetadata: AlternativeRouteMetadata
+    ): RouteProgressData {
+        val legIndex: Int
+        val routeGeometryIndex: Int
+        val legGeometryIndex: Int?
+        val primaryFork = alternativeMetadata.forkIntersectionOfPrimaryRoute
+        val alternativeFork = alternativeMetadata.forkIntersectionOfAlternativeRoute
+        if (primaryRouteProgressData.routeGeometryIndex < primaryFork.geometryIndexInRoute) {
+            val legIndexDiff = primaryFork.legIndex - alternativeFork.legIndex
+            legIndex = primaryRouteProgressData.legIndex - legIndexDiff
+            val routeGeometryIndexDiff =
+                primaryFork.geometryIndexInRoute - alternativeFork.geometryIndexInRoute
+            routeGeometryIndex =
+                primaryRouteProgressData.routeGeometryIndex - routeGeometryIndexDiff
+            val legGeometryIndexDiff =
+                primaryFork.geometryIndexInLeg - alternativeFork.geometryIndexInLeg
+            legGeometryIndex = primaryRouteProgressData.legGeometryIndex?.let {
+                it - legGeometryIndexDiff
+            }
+        } else {
+            legIndex = alternativeFork.legIndex
+            routeGeometryIndex = alternativeFork.geometryIndexInRoute
+            legGeometryIndex = alternativeFork.geometryIndexInLeg
+        }
```

Comment:
does it mean, that we have passed alternative, but didn't remove it from the tracking routes, we will be refreshing it starting from fork point?

Reply:
Exactly.